### PR TITLE
Add likes and comments API

### PR DESCRIPTION
**Pull request by ghost**
Added a mock recommendation api.
Added missing fields in file schema

### CODE REVIEW

File: resources/file/fileSchema.js [hunk 1]
```diff
@@ -36,6 +37,20 @@ layerSchema.statics.newTestLayer = function () {
 
 const fileSchema = Schema({
   authorUsername: { type: String, required: true, index: true },
+  name: { type: String, required: [true, 'Name is required'] },
+  type: { type: String, required: true, enum: ['map', 'tileset'], index: true },
+  description: { type: String, required: true },
```
Comment (lincolnnguyen18) [inline resources/file/fileSchema.js]: description shouldnt be required

File: resources/file/fileSchema.js [hunk 2]
```diff
@@ -76,20 +79,27 @@ fileSchema.statics.newTestFile = async function (authorUsername) {
   rootLayer.layers = createRandomTree(3);
   await rootLayer.save();
 
+  const likeCnt = _.random(0, 100);
+  const comCnt = _.random(0, 20);
```
Comment (lincolnnguyen18) [inline resources/file/fileSchema.js]: don't abbreviate, use full likeCount name and include in return object without colon

File: package.json [hunk 3]
```diff
@@ -19,7 +19,8 @@
     "express": "^4.18.1",
     "jsonwebtoken": "^8.5.1",
     "lodash": "^4.17.21",
-    "mongoose": "^6.6.0"
+    "mongoose": "^6.6.0",
+    "random-sentence": "^1.0.4"
```
Comment (lincolnnguyen18) [inline package.json]: don't use random sentence use faker https://fakerjs.dev/api/lorem.html#sentence
keep dependencies to minimum

File: resources/file/fileSchema.js [hunk 4]
```diff
@@ -101,7 +111,30 @@ layerSchema.statics.deleteTestLayers = async function () {
   await this.deleteMany({ name: /test_root_layer/ });
 };
 
+const newTestComments = function (count) {
+  const comments = [];
+  for (let i = 0; i < count; i++) {
+    comments.push({
+      username: faker.random.words(_.random(1, 5)),
```
Comment (lincolnnguyen18) [inline resources/file/fileSchema.js]: username should not be random, should be valid usernames, pass in array of usernames as parameter, same with likes

File: resources/file/fileController.js [hunk 5]
```diff
@@ -254,4 +256,26 @@ async function addCommentToFile (req, res) {
...
+  const findQuery = {};
+
+  findQuery.$text = { $search: file.tags };
+
+  const files = await File
+    .find(findQuery)
+    .limit(limit ?? 10)
+    .select(viewFileFields.join(' '))
+    .catch(() => []);
+
```
Comment (lincolnnguyen18) [inline resources/file/fileController.js]: Can you add random files that are not yet already in the files array until the specified limit is reached? If there are no files or no more unique files then just return an empty array. Also make sure the file specified in the id is not part of the recommended files returned.

**Review by lincolnnguyen18 — CHANGES_REQUESTED** [review]
Recommenation endpoint is a good idea!

File: resources/file/fileController.js (hunk 5 above)
Comment (ghost) [reply resources/file/fileController.js]: For now, the recommendation api is only a mock up. I'll return a random list of files. If I do find time after static assets are done. I'll work on the algorithm

**Comment by ghost** [comment]
Recommendation api is not working. I do not expect it to work in this branch.
I'll add the algorithm after the mock up for recommendation and comments are done.

**Review by lincolnnguyen18 — DISMISSED** [review]

**Review by lincolnnguyen18 — APPROVED** [review]

**Pull request merged**